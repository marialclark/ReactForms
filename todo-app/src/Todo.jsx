import './styles/Todo.css';

const Todo = ({ id, task, removeTodo }) => {
  return (
    <div className="Todo">
      <span>{task}</span>
      <button onClick={() => removeTodo(id)}>X</button>
    </div>
  );
};

export default Todo;