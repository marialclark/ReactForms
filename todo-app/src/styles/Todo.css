/* Styling for individual Todo items */
.Todo {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 5px 10px;
  border: 1px solid #ccc;
  border-radius: 3px;
  background-color: white;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
}

.Todo button {
  background-color: red;
  color: white;
  border: none;
  padding: 3px 8px;
  border-radius: 50%;
  cursor: pointer;
  font-size: 12px;
}

.Todo button:hover {
  background-color: darkred;
}
