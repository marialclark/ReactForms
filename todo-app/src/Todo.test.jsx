import { render, screen, fireEvent } from '@testing-library/react';
import Todo from './Todo';
import { vi } from 'vitest'; // For mocking

describe('Todo Component', () => {
  // Smoke Test
  it('renders without crashing', () => {
    render(<Todo id="test-id" task="Test Task" removeTodo={() => {}} />);
  });

  // Snapshot Test
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Todo id="test-id" task="Test Task" removeTodo={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Critical Business Logi: Remove Todo
  it('calls removeTodo with the correct id when the button is clicked', () => {
    const mockRemoveTodo = vi.fn(); // Mock the removeTodo function
    render(<Todo id="test-id" task="Test Task" removeTodo={mockRemoveTodo} />);

    const button = screen.getByText('X');
    fireEvent.click(button);

    expect(mockRemoveTodo).toHaveBeenCalledTimes(1);
    expect(mockRemoveTodo).toHaveBeenCalledWith('test-id');
  });
});
