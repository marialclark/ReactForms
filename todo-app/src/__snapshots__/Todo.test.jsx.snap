// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Todo Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span>
      Test Task
    </span>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
